components/filter/test/filter-tags.vdiff.js

```diff
@@ -0,0 +1,143 @@
+			<d2l-filter-dimension-set-value ?selected="${moreSelected}" text="Value 3" key="3"></d2l-filter-dimension-set-value>
+		</d2l-filter-dimension-set>
+	</d2l-filter>
+`;
+
+const tagsSingleFilter = html`
+	<div style="display: flex; justify-content: space-between;">
+		<d2l-filter-tags filter-ids="filter" style="align-self: center; position: relative; width: 100%;"></d2l-filter-tags>
+		${singleFilter}
+	</div>
+`;
+const tagsFlexEndSingleFilter = html`
+	<div>
+		${singleFilter}
+		<div id="capture" style="display: flex; justify-content: flex-end;">
+			<d2l-filter-tags filter-ids="filter"></d2l-filter-tags>
+		</div>
+	</div>
+`;
+const tagsTwoFilters = html`
+	<div style="display: flex; justify-content: space-between;">
+		<d2l-filter-tags filter-ids="filter-1 filter-2" style="align-self: center; position: relative; width: 100%;"></d2l-filter-tags>
+		${twoFilters()}
+	</div>
+`;
+
+const tagsTwoFiltersMoreSelected = html`
+	<div style="display: flex; justify-content: space-between;">
+		<d2l-filter-tags filter-ids="filter-1 filter-2" style="align-self: center; position: relative; width: 100%;"></d2l-filter-tags>
+		${twoFilters(true)}
+	</div>
+`;
+
+describe('filter-tags', () => {
+	[
+		{ name: 'two-filters', template: tagsTwoFilters },
+		{ name: 'flex-end', template: tagsFlexEndSingleFilter, selector: '#capture' },
+		{ name: 'basic-rtl', rtl: true, template: tagsSingleFilter },
+		{ name: 'two-filters-rtl', rtl: true, template: tagsTwoFilters }
+	].forEach(({ name, template, rtl, selector }) => {
+		it(name, async() => {
+			let elem = await fixture(template, { rtl, viewport: { width: 1700 } });
+			if (selector) elem = elem.querySelector(selector);
+			await expect(elem).to.be.golden();
+		});
+	});
+
+	[1500, 980, 969, 601, 599, 400, 320].forEach((width) => {
+		describe('basic width', () => {
+			let elem;
+			beforeEach(async() => {
+				elem = await fixture(tagsSingleFilter, { viewport: { width: width + 216 } }); // account for filter and page padding
+			});
+
+			it(`${width}`, async() => {
+				await expect(elem).to.be.golden();
+			});
+
+			it(`${width} show more button clicked`, async() => {
+				const filterTags = elem.querySelector('d2l-filter-tags');
+				const tagList = filterTags.shadowRoot.querySelector('d2l-tag-list');
+				const showMoreButton = tagList.shadowRoot.querySelector('.d2l-tag-list-button');
+
+				if (showMoreButton) await clickElem(showMoreButton);
+				await expect(elem).to.be.golden();
+			});
+		});
+	});
+
+	describe('clearable behavior', () => {
+		it('deleting the last item', async() => {
+			const elem = await fixture(tagsSingleFilter, { viewport: { width: 1700 } });
+			const filterTags = elem.querySelector('d2l-filter-tags');
+			const items = filterTags.shadowRoot.querySelectorAll('d2l-tag-list-item');
+			const deleteButton = items[6].shadowRoot.querySelector('d2l-button-icon');
+
+			await clickElem(deleteButton);
+			await aTimeout('300'); // clearing has a 200ms delay
```

Comment:
These delays are unfortunate but unavoidable I guess eh.

Reply:
Hmm, I can try playing around with sinon timer stuff

Reply:
Actually, I'm gonna merge this as is and come back to this if I have time. I'm running into issues with mocking the timers affecting the other stuff that needs to run in the test. We could also consider allowing this timeout to be configurable for tests